### ISSUE TITLE
libc: src/math/fma.cpp does not compile with clang 15.0.0 LATEST

### ISSUE THREAD

**Issue by ErichLoew** [llvm/llvm-project#55887]
Date:     20220605
OS:       OpenSUSE Leap 15.2
Kernel:   5.18.1
Compiler: clang version 15.0.0 (https://github.com/llvm/llvm-project.git 9e6bbd051fb5fe8e9cb7e525c17d3099cce382ab)
File:     llvm/libc/src/math/fma.cpp
Issue:  In original code the reference to namespace fputil is not present and this triggers a clang++ compile error.
```console
/home/SETUP/clang/LATEST/llvm/libc/src/math/fma.cpp:19:20: error: no matching function for call to 'fma'
            return fputil::fma(x, y, z);
                   ^~~~~~~~~~~
/home/SETUP/clang/LATEST/llvm/libc/src/__support/FPUtil/FMA.h:33:66: note: candidate template ignored: substitution failure [with T = double]: implicit instantiation of undefined template '__llvm_libc::cpp::EnableIf<false, double>'
static inline cpp::EnableIfType<cpp::IsSame<T, float>::Value, T> fma(T x, T y,
```

The following code experiment compiles

```cpp
namespace __llvm_libc
{
    namespace fputil
    {
        LLVM_LIBC_FUNCTION(double, fma, (double const x, double const y, double const z))
        {
            return fputil::fma(x, y, z);
        }
    } // namespace fputil
} // namespace __llvm_libc
```

**Comment by lntue** [comment]
It looks like the code in `libc/src/__support/FPUtil/FMA.h` is before we added full support for `fma(double, double, double)` for all targets besides aarch64 and x86-64 with FMA in https://reviews.llvm.org/rGee8992770754d186c9217d35cccd339efc55a36b

So it seems to be one of the following possibilities:
- The target does not have FMA instructions
- We don't support FMA instructions for the target yet
- Our FMA flags detection in cmake and/or `libc/src/__support/FPUtil/FMA.h` somehow miss the target

@ErichLoew : Do you mind checking if the problem still persist with HEAD?  Thanks,

**Comment by ErichLoew** [comment]
I‘ll check, very thx for advise!Mit freundlichen Grüßen,Erich LöwAm 01.12.2022 um 17:00 schrieb lntue ***@***.***>:﻿
It looks like the code in libc/src/__support/FPUtil/FMA.h is before we added full support for fma(double, double, double) for all targets besides aarch64 and x86-64 with FMA in https://reviews.llvm.org/rGee8992770754d186c9217d35cccd339efc55a36b
So it seems to be one of the following possibilities:

The target does not have FMA instructions
We don't support FMA instructions for the target yet
Our FMA flags detection in cmake and/or libc/src/__support/FPUtil/FMA.h somehow miss the target

@ErichLoew : Do you mind checking if the problem still persist with HEAD?  Thanks,

**Comment by lntue** [comment]
I'll close this issue for now.  @ErichLoew : feel free to re-open or create a new issue for me if you still have problem with FMA functions.
